File: id/src/main/java/com/simprints/id/domain/fingerprint/ScanConfig.java
package com.simprints.libcommon;

import com.simprints.libsimprints.FingerIdentifier;

import java.util.Arrays;

@SuppressWarnings("unused")
public class ScanConfig {

    private final FingerConfig[] fingerConfigs;
    private final int[] priorities;
    private final int[] orders;

    /**
     * Constructor.
     * Constructs a config where all the fingers are optional and have priority 0
     */
    public ScanConfig()
    {
        this.fingerConfigs = new FingerConfig[FingerIdentifier.values().length];
        Arrays.fill(this.fingerConfigs, FingerConfig.OPTIONAL);
        this.priorities = new int[FingerIdentifier.values().length];
        Arrays.fill(this.priorities, 0);
        this.orders = new int[FingerIdentifier.values().length];
        Arrays.fill(this.orders, 0);
    }

    public void set(FingerIdentifier id, FingerConfig config, int priority, int order)
    {
        fingerConfigs[id.ordinal()] = config;
        this.priorities[id.ordinal()] = priority;
        this.orders[id.ordinal()] = order;
    }

    public FingerConfig get(FingerIdentifier id)
    {
        return fingerConfigs[id.ordinal()];
    }

    public int getPriority(FingerIdentifier id)
    {
        return priorities[id.ordinal()];
    }

    public int getOrder(FingerIdentifier id) {
        return orders[id.ordinal()];
    }
}

File: id/src/main/java/com/simprints/id/domain/fingerprint/ScanConfig.kt
package com.simprints.id.domain.fingerprint


import com.simprints.id.FingerIdentifier

class ScanConfig {

    val fingerConfigs: Map<FingerIdentifier, ScanConfigFingerEntry> =
        FingerIdentifier.values().associate { Pair(it, ScanConfigFingerEntry()) }

    operator fun get(id: FingerIdentifier): FingerConfig =
        fingerConfigs[id]?.let { it.config }
            ?: throw IllegalArgumentException("No FingerConfig associated to id")

    fun getPriority(id: FingerIdentifier): Int =
        fingerConfigs[id]?.let { it.priority }
            ?: throw IllegalArgumentException("No priority associated to id")

    fun getOrder(id: FingerIdentifier): Int =
        fingerConfigs[id]?.let { it.order }
            ?: throw IllegalArgumentException("No priority associated to id")
}
